File: src/test/java/me/piclane/logview/procedure/text/CharsetDetectorTest.java
package me.piclane.logview.procedure.text;

import org.junit.Test;

import java.nio.charset.Charset;
import java.nio.charset.StandardCharsets;
import java.nio.file.Path;
import java.nio.file.Paths;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.*;

public class CharsetDetectorTest {
    @Test
    public void test_utf8() throws Exception {
        Path path = Paths.get(CharsetDetectorTest.class.getResource("sample/text_utf8.txt").toURI());
        Charset actual = CharsetDetector.detect(path);
        assertThat(actual, is(equalTo(StandardCharsets.UTF_8)));
    }

    @Test
    public void test_utf16le() throws Exception {
        Path path = Paths.get(CharsetDetectorTest.class.getResource("sample/text_utf16le.txt").toURI());
        Charset actual = CharsetDetector.detect(path);
        assertThat(actual, is(equalTo(StandardCharsets.UTF_16LE)));
    }

    @Test
    public void test_utf16be() throws Exception {
        Path path = Paths.get(CharsetDetectorTest.class.getResource("sample/text_utf16be.txt").toURI());
        Charset actual = CharsetDetector.detect(path);
        assertThat(actual, is(equalTo(StandardCharsets.UTF_16BE)));
    }

    @Test
    public void test_sjis() throws Exception {
        Path path = Paths.get(CharsetDetectorTest.class.getResource("sample/text_sjis.txt").toURI());
        Charset actual = CharsetDetector.detect(path);
        assertThat(actual, is(equalTo(Charset.forName("windows-31j"))));
    }

    @Test
    public void test_eucjp() throws Exception {
        Path path = Paths.get(CharsetDetectorTest.class.getResource("sample/text_eucjp.txt").toURI());
        Charset actual = CharsetDetector.detect(path);
        assertThat(actual, is(equalTo(Charset.forName("EUC-JP"))));
    }
}

File: src/test/kotlin/com/xxuz/piclane/logview/procedure/text/CharsetDetectorTest.kt
package com.xxuz.piclane.logview.procedure.text

import com.xxuz.piclane.logview.procedure.text.CharsetDetector.detect
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.Test
import java.nio.charset.Charset
import java.nio.charset.StandardCharsets
import java.nio.file.Paths

class CharsetDetectorTest {
    @Test
    fun test_utf8() {
        val path = Paths.get(CharsetDetectorTest::class.java.getResource("/CharsetDetectorTest/text_utf8.txt").toURI())
        val actual = detect(path)
        MatcherAssert.assertThat(actual, Matchers.`is`(Matchers.equalTo(StandardCharsets.UTF_8)))
    }

    @Test
    fun test_utf16le() {
        val path = Paths.get(CharsetDetectorTest::class.java.getResource("/CharsetDetectorTest/text_utf16le.txt").toURI())
        val actual = detect(path)
        MatcherAssert.assertThat(actual, Matchers.`is`(Matchers.equalTo(StandardCharsets.UTF_16LE)))
    }

    @Test
    fun test_utf16be() {
        val path = Paths.get(CharsetDetectorTest::class.java.getResource("/CharsetDetectorTest/text_utf16be.txt").toURI())
        val actual = detect(path)
        MatcherAssert.assertThat(actual, Matchers.`is`(Matchers.equalTo(StandardCharsets.UTF_16BE)))
    }

    @Test
    fun test_sjis() {
        val path = Paths.get(CharsetDetectorTest::class.java.getResource("/CharsetDetectorTest/text_sjis.txt").toURI())
        val actual = detect(path)
        MatcherAssert.assertThat(actual, Matchers.`is`(Matchers.equalTo(Charset.forName("windows-31j"))))
    }

    @Test
    fun test_eucjp() {
        val path = Paths.get(CharsetDetectorTest::class.java.getResource("/CharsetDetectorTest/text_eucjp.txt").toURI())
        val actual = detect(path)
        MatcherAssert.assertThat(actual, Matchers.`is`(Matchers.equalTo(Charset.forName("EUC-JP"))))
    }
}
